metric/bigtable/src/main/java/com/spotify/heroic/metric/bigtable/BigtableBackend.java

```diff
@@ -447,30 +449,37 @@ public Statistics getStatistics() {
                 cell -> p.deserialize(cell.getQualifier(), cell.getValue());
 
             QueryTrace.NamedWatch fs = QueryTrace.watch(FETCH_SEGMENT);
-            final AsyncFuture<List<FlatRow>> readRows =
-              client.readRows(
-                  table,
-                  ReadRowsRequest.builder()
-                      .range(new RowRange(Optional.of(p.rowKeyStart), Optional.of(p.rowKeyEnd)))
-                      .filter(
-                          RowFilter.chain(
-                              Arrays.asList(
-                                  RowFilter.newColumnRangeBuilder(p.columnFamily)
-                                      .startQualifierOpen(p.startQualifierOpen)
-                                      .endQualifierClosed(p.endQualifierClosed)
-                                      .build(),
-                                  RowFilter.onlyLatestCell())))
-                      .build()
-              ).onDone(new EndSpanFutureReporter(readRowsSpan));
 
-            fetches.add(readRows.directTransform(result -> {
-                final Span transformSpan = tracer.spanBuilderWithExplicitParent(
-                    "bigtable.transform", readRowsSpan).startSpan();
-                transformSpan.putAttribute("rowsReturned", longAttributeValue(result.size()));
+            final AsyncFuture<List<FlatRow>> readRows;
+            try (Scope ignored = tracer.withSpan(fetchBatchSpan)) {
+                readRows =
+                    client
+                        .readRows(
+                            table,
+                            ReadRowsRequest.builder()
+                                .range(new RowRange(Optional.of(p.rowKeyStart), Optional.of(p.rowKeyEnd)))
+                                .filter(
+                                    RowFilter.chain(
+                                        Arrays.asList(
+                                            RowFilter.newColumnRangeBuilder(p.columnFamily)
+                                                .startQualifierOpen(p.startQualifierOpen)
+                                                .endQualifierClosed(p.endQualifierClosed)
+                                                .build(),
+                                            RowFilter.onlyLatestCell())))
+                                .build())
+                        .onDone(new EndSpanFutureReporter(readRowsSpan));
+            }
 
+            final AtomicBoolean foundResourceIdentifier = new AtomicBoolean(false);
```

Comment:
Does this need to be an AtomicBoolean instead of a plain Boolean? Its threaded access, but it can only every get flipped to `true` so should be fine to use across threads. Not sure if it matters/has performance impact.